### PR TITLE
More functionalities for collection and path

### PR DESCRIPTION
**Pull request by ndhuutai**
Added:
_save a path (touched both front and backend)
_fetch all paths (basic)

### CODE REVIEW

File: client/src/components/collection-expand-view/collection-expand-view.component.jsx [hunk 1]
```diff
@@ -6,15 +6,19 @@ import { useHistory } from 'react-router-dom';
 import PathEditorContext from '../../contexts/path-editor-context';
 import { addCollection } from '../../actions/path-editor';
 import styles from './collection-expand-view.css';
+// import { collection } from '../../../../server/models/userpath';
```
Comment (robertxluo) [inline client/src/components/collection-expand-view/collection-expand-view.component.jsx:9]: Remove this comment

File: server/routes/api/userpaths.js [hunk 2]
```diff
@@ -5,35 +5,35 @@ const express = require('express');
...
-      console.log(err);
-      res.status(500).send(`Server error: ${err._message}`);
-    }
-  });
+router.get('/', async (req, res) => {
+  try {
+    const allpaths = await UserPath.find({}).exec();
+    res.json(allpaths);
+  } catch (e) {
+    console.log(e);
```
Comment (robertxluo) [inline server/routes/api/userpaths.js:13]: Remove this console.log

File: server/routes/api/userpaths.js [hunk 3]
```diff
@@ -5,35 +5,35 @@ const express = require('express');
...
+    if (tags) userPathDetails.tags = tags;
+    if (likes) userPathDetails.likes = likes;
+    if (collections) userPathDetails.collections = collections;
+    if (mainPath) userPathDetails.mainPath = mainPath;
+    if (completed) userPathDetails.completed = completed;
+    const userPath = new UserPath(userPathDetails);
+    await userPath.save();
+    res.json(userPath);
+  } catch (err) {
+    console.log(err);
```
Comment (robertxluo) [inline server/routes/api/userpaths.js:34]: Remove this console.log

File: client/src/components/collection-expand-view/collection-expand-view.component.jsx (hunk 1 above)
Comment (ndhuutai) [reply client/src/components/collection-expand-view/collection-expand-view.component.jsx]: done

File: server/routes/api/userpaths.js (hunk 2 above)
Comment (ndhuutai) [reply server/routes/api/userpaths.js]: done

File: server/routes/api/userpaths.js (hunk 3 above)
Comment (ndhuutai) [reply server/routes/api/userpaths.js]: done

**Pull request merged**